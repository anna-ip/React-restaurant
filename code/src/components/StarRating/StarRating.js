import React from "react";
import { FaStar } from "react-icons/fa";
import styles from "./StarRating.module.scss";

const StarRating = ({ handleRating, rating }) => {
  const arr = [...Array(5)];

  return (
    <div className={styles.container}>
      {arr.map((star, index) => {
        const ratingValue = index + 1;
        return (
          <label key={index}>
            <input
              type="radio"
              name="rating"
              value={ratingValue}
              onClick={handleRating}
            />
            <FaStar
              className={
                ratingValue <= rating ? styles.yellowStar : styles.star
              }
            />
          </label>
        );
      })}
    </div>
  );
};

export default StarRating;
