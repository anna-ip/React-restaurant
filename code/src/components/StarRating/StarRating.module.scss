@import "../../styles/variables";

.container{
  display: flex;
  flex-direction: row;
  
  input[type="radio"] {
    display: none;
  }
  .star {
    cursor: pointer;
    transition: color 200ms;
    font-size: 1.2rem;
    color: $base-color;
  }
  
  .yellowStar {
    cursor: pointer;
    transition: color 200ms;
    font-size: 1.2rem;
    color: #ffe100;
  }
}
