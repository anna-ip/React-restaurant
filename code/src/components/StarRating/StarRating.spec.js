import React from "react";
import { shallow } from "enzyme";
import StarRating from "./StarRating";

describe("<StarRating/>", () => {
  test("snapshot test", () => {
    const wrapper = shallow(<StarRating />);
    expect(wrapper).toMatchSnapshot();
  });
});
