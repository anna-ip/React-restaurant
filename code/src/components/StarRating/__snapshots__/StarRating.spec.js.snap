// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating/> snapshot test 1`] = `
<div
  className="container"
>
  <label
    key="0"
  >
    <input
      name="rating"
      type="radio"
      value={1}
    />
    <FaStar
      className="star"
    />
  </label>
  <label
    key="1"
  >
    <input
      name="rating"
      type="radio"
      value={2}
    />
    <FaStar
      className="star"
    />
  </label>
  <label
    key="2"
  >
    <input
      name="rating"
      type="radio"
      value={3}
    />
    <FaStar
      className="star"
    />
  </label>
  <label
    key="3"
  >
    <input
      name="rating"
      type="radio"
      value={4}
    />
    <FaStar
      className="star"
    />
  </label>
  <label
    key="4"
  >
    <input
      name="rating"
      type="radio"
      value={5}
    />
    <FaStar
      className="star"
    />
  </label>
</div>
`;
